packages/ckeditor5-image/src/imageresize/imageresizehandles.js

```diff
@@ -74,13 +78,53 @@ export default class ImageResizeHandles extends Plugin {
 
 			resizer.on( 'updateSize', () => {
 				if ( !widget.hasClass( 'image_resized' ) ) {
-					editor.editing.view.change( writer => {
+					editingView.change( writer => {
 						writer.addClass( 'image_resized', widget );
 					} );
 				}
 			} );
 
+			this._hideResizerUntilImageIsLoaded( resizer, widget );
+
 			resizer.bind( 'isEnabled' ).to( this );
 		}, { priority: 'low' } );
 	}
+
+	/**
+	 * Function to add listeners that will hide the resizer frame until the image is loaded,
+	 * to avoid [resize frame flashing when the image is not yet loaded](https://github.com/ckeditor/ckeditor5/issues/8088).
+	 *
+	 * @private
+	 * @param {module:widget/widgetresize/resizer~Resizer} resizer
+	 * @param {module:engine/view/containerelement~ContainerElement} widget
+	 */
+	_hideResizerUntilImageIsLoaded( resizer, widget ) {
+		// Given that #7548 is fixed, this logic might no longer be needed.
+		const editor = this.editor;
+		const editingView = editor.editing.view;
+
+		editingView.change( writer => {
+			writer.addClass( 'image_resizer_loading', widget );
+		} );
+
+		editingView.document.on( 'imageLoaded', imageLoadCallback );
+
+		function imageLoadCallback( evt, domEvent ) {
+			const handleHost = resizer._getHandleHost();
+
+			if ( domEvent.target.isSameNode( handleHost ) ) {
+				editingView.change( writer => {
+					if ( editor.plugins.get( WidgetResize )._visibleResizer == resizer ) {
```

Comment:
Let's add \_visibleResizer to the public API. It's already used cross-package for testing purposes so it should be public already.